app/views/partials/error_summary.scala.html

```diff
@@ -22,7 +22,7 @@ <h2 class="heading-medium error-summary-heading" id="error-summary-heading">
         </h2>
         <ul role="list" class="error-summary-list">
             @for((error: FormError, index: Int) <- errors.zipWithIndex) {
-                <li><a href="#@{error.key}" tabindex="@startAtTabIndex.map(x => x+2+index)">@messages(error.message, error.args:_*)</a></li>
+                <li><a href="#@error.key.replaceAll("""[\.\[\]]""", "_")" tabindex="@startAtTabIndex.map(x => x+2+index)">@messages(error.message, error.args:_*)</a></li>
```

Comment:
Could you explain what is going on in this bit? 😛

Reply:
We need the href in the error messages to point to the correct input by the input's id. The id for an input form is being constructed in the component with '_' e.g. 'date_day' but the error key for it would be 'day.day' so we are replacing the '.' with '_' of the error key in the href